### ISSUE TITLE
Postgres: SQLFluff failing to parse JSON expressions with #> and #>> operators

### ISSUE THREAD

**Issue by derickl** [sqlfluff/sqlfluff#2033]
## Expected Behaviour
SQLFluff should parse JSON expressions that use the following operators `#>`, `#>>` in Postgres

## Observed Behaviour

```shell
sqlfluff lint --dialect postgres -n sample.sql

== [postgres/sample.sql] FAIL                                                                                                                                                                                  
L:   1 | P:  16 |  PRS | Couldn't find closing bracket for opening bracket.
```

```shell
sqlfluff parse --dialect postgres -n sample.sql

...Failed to Parse...
==== parsing violations ====
L:   1 | P:  16 |  PRS | Couldn't find closing bracket for opening bracket.
```

## Steps to Reproduce

1. Create sample.sql using the following query
```sql
SELECT COALESCE(doc#>>'{fields}','') AS field
FROM mytable
WHERE doc ->> 'some_field' = 'some_value';
```

2. sqlfluff lint --dialect postgres sample.sql
3. sqlfluff parse --dialect postgres -n sample.sql

## Dialect
`postgres`

## Version
`sqlfluff, version 0.8.2 and master`
`Python 3.8.22`

## Other information
If we are to remove the coalesce operation and look at the parse tree, we can see that everything after the `#` character is treated as a comment.

```sql
SELECT doc#>>'{fields}' AS field
FROM mytable
WHERE doc ->> 'some_field' = 'some_value';
```

```shell
 sqlfluff parse --dialect=postgres -n postgres/sample.sql
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        select_statement:
[L:  1, P:  1]      |            select_clause:
[L:  1, P:  1]      |                keyword:                                      'SELECT'
[L:  1, P:  7]      |                [META] indent:
[L:  1, P:  7]      |                whitespace:                                   ' '
[L:  1, P:  8]      |                select_clause_element:
[L:  1, P:  8]      |                    column_reference:
[L:  1, P:  8]      |                        identifier:                           'doc'
[L:  1, P: 11]      |            comment:                                          "#>>'{fields}' AS field"
[L:  1, P: 33]      |            newline:                                          '\n'
[L:  2, P:  1]      |            [META] dedent:
[L:  2, P:  1]      |            from_clause:
[L:  2, P:  1]      |                keyword:                                      'FROM'
[L:  2, P:  5]      |                whitespace:                                   ' '
[L:  2, P:  6]      |                from_expression:
[L:  2, P:  6]      |                    [META] indent:
[L:  2, P:  6]      |                    from_expression_element:
[L:  2, P:  6]      |                        table_expression:
[L:  2, P:  6]      |                            table_reference:
[L:  2, P:  6]      |                                identifier:                   'mytable'
[L:  2, P: 13]      |                    [META] dedent:
[L:  2, P: 13]      |            newline:                                          '\n'
[L:  3, P:  1]      |            where_clause:
[L:  3, P:  1]      |                keyword:                                      'WHERE'
[L:  3, P:  6]      |                [META] indent:
[L:  3, P:  6]      |                whitespace:                                   ' '
[L:  3, P:  7]      |                expression:
[L:  3, P:  7]      |                    column_reference:
[L:  3, P:  7]      |                        identifier:                           'doc'
[L:  3, P: 10]      |                    whitespace:                               ' '
[L:  3, P: 11]      |                    binary_operator:                          '->>'
[L:  3, P: 14]      |                    whitespace:                               ' '
[L:  3, P: 15]      |                    literal:                                  "'some_field'"
[L:  3, P: 27]      |                    whitespace:                               ' '
[L:  3, P: 28]      |                    comparison_operator:                      '='
[L:  3, P: 29]      |                    whitespace:                               ' '
[L:  3, P: 30]      |                    literal:                                  "'some_value'"
[L:  3, P: 42]      |                [META] dedent:
[L:  3, P: 42]      |    statement_terminator:                                     ';'
[L:  3, P: 43]      |    newline:                                                  '\n'
```

If we switch to using this `->>` operator, SQLFluff generates the correct parse tree

```sql
SELECT doc ->> '{fields}' AS field
FROM mytable
WHERE doc ->> 'some_field' = 'some_value';
```

```shell
sqlfluff parse --dialect=postgres -n postgres/_formview_vht_consumption_log.sql
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        select_statement:
[L:  1, P:  1]      |            select_clause:
[L:  1, P:  1]      |                keyword:                                      'SELECT'
[L:  1, P:  7]      |                [META] indent:
[L:  1, P:  7]      |                whitespace:                                   ' '
[L:  1, P:  8]      |                select_clause_element:
[L:  1, P:  8]      |                    expression:
[L:  1, P:  8]      |                        column_reference:
[L:  1, P:  8]      |                            identifier:                       'doc'
[L:  1, P: 11]      |                        whitespace:                           ' '
[L:  1, P: 12]      |                        binary_operator:                      '->>'
[L:  1, P: 15]      |                        whitespace:                           ' '
[L:  1, P: 16]      |                        literal:                              "'{fields}'"
[L:  1, P: 26]      |                    whitespace:                               ' '
[L:  1, P: 27]      |                    alias_expression:
[L:  1, P: 27]      |                        keyword:                              'AS'
[L:  1, P: 29]      |                        whitespace:                           ' '
[L:  1, P: 30]      |                        identifier:                           'field'
[L:  1, P: 35]      |            newline:                                          '\n'
[L:  2, P:  1]      |            [META] dedent:
[L:  2, P:  1]      |            from_clause:
[L:  2, P:  1]      |                keyword:                                      'FROM'
[L:  2, P:  5]      |                whitespace:                                   ' '
[L:  2, P:  6]      |                from_expression:
[L:  2, P:  6]      |                    [META] indent:
[L:  2, P:  6]      |                    from_expression_element:
[L:  2, P:  6]      |                        table_expression:
[L:  2, P:  6]      |                            table_reference:
[L:  2, P:  6]      |                                identifier:                   'mytable'
[L:  2, P: 13]      |                    [META] dedent:
[L:  2, P: 13]      |            newline:                                          '\n'
[L:  3, P:  1]      |            where_clause:
[L:  3, P:  1]      |                keyword:                                      'WHERE'
[L:  3, P:  6]      |                [META] indent:
[L:  3, P:  6]      |                whitespace:                                   ' '
[L:  3, P:  7]      |                expression:
[L:  3, P:  7]      |                    column_reference:
[L:  3, P:  7]      |                        identifier:                           'doc'
[L:  3, P: 10]      |                    whitespace:                               ' '
[L:  3, P: 11]      |                    binary_operator:                          '->>'
[L:  3, P: 14]      |                    whitespace:                               ' '
[L:  3, P: 15]      |                    literal:                                  "'some_field'"
[L:  3, P: 27]      |                    whitespace:                               ' '
[L:  3, P: 28]      |                    comparison_operator:                      '='
[L:  3, P: 29]      |                    whitespace:                               ' '
[L:  3, P: 30]      |                    literal:                                  "'some_value'"
[L:  3, P: 42]      |                [META] dedent:
[L:  3, P: 42]      |    statement_terminator:                                     ';'
[L:  3, P: 43]      |    newline:                                                  '\n'
```

**Comment by jpy-git** [comment]
@derickl Good breakdown on this issue 😄 

I can see that the hash comment is set in the ansi dialect (which postgres inherits) but we should overwrite this behaviour for postgres as it doesn't have hash comments. I can also see that the json operators have previously been added to postgres so I'm hoping once we remove the hash comment from postgres this should start parsing.
![image](https://user-images.githubusercontent.com/80432516/144691114-594e2463-ff1f-4380-8c11-e4e503a18db5.png)

Will take a look tomorrow 👍
